Dockerfile

```diff
@@ -1,10 +1,31 @@
 FROM circleci/node:10.11
 
-WORKDIR /opt/sinopia_editor/
+# Allow build-time arguments (for, e.g., docker-compose)
+ARG SPOOF_SINOPIA_SERVER
+ARG TRELLIS_BASE_URL
+ARG DEFAULT_PROFILE_SCHEMA_VERSION
+ARG SINOPIA_GROUP
+ARG SINOPIA_URI
+ARG AWS_COGNITO_DOMAIN
+ARG COGNITO_CLIENT_ID
+
+# This is the directory the user in the circleci/node image can write to
+WORKDIR /home/circleci
 
 # Everything that isn't in .dockerignore ships
 COPY . .
 
+# Allow circleci user to run npm build
+USER root
+RUN /bin/bash -c 'chown -R circleci dist'
+
+# Run app using non-privileged account
+USER circleci
+
+# Build the app *within* the container because environment variables are fixed at build-time
+RUN npm install
+RUN npm run build
```

Comment:
If we do not build the app when building the container, we cannot inject environment variables via container machinery, because environment variables (in `process.env`) are fixed at build time.

Reply:
Thanks @mjgiarlo for investigating this issue and the fixes in this PR. I have to keep reminding myself that React applications run client-side and not on the server.